File: ruby/word-count/_iterations/1/word_count.rb
class Phrase
  attr_reader :word_count

  def initialize(phrase)
    @word_count = phrase.scan(/\w+(?:'\w+)?/).reduce(Hash.new(0)) do |hash, word|
      hash[word.downcase] = hash[word.downcase] + 1
      hash
    end
  end
end

File: kotlin/word-count/_iterations/1/src/main/kotlin/WordCount.kt
object WordCount {
    private val WORD = """\w(?!\w)|\w[\w']*\w""".toRegex()

    fun phrase(input: String): Map<String, Int> {
        return WORD.findAll(input.toLowerCase()).groupingBy { it.value }.eachCount()
    }
}
